### PR TITLE
feat: migrate bitcoin generate txs, closes LEA-1735

### PR DESCRIPTION
**Pull request by fbwoolf**
This PR migrates code to generate the unsigned psbt for btc sends.

I did some refactoring here using generic types so we can extend our form context to accommodate diff form data in each form flow. I needed to do this to handle loading `feeRates` and `utxos` for bitcoin specifically. I spent a lot of time trying to rethink this and hope it is a good solution. I've created a few more loaders to get rid of some undefined returns we had in the code I ported over from the extension.

I've also spent some time making sure to write tests that were missing, and refactoring bitcoin fees code as I migrated it over.

As mentioned in Slack, I'd like to standardize our `Fees` model type after this work.

### CODE REVIEW

**Review by kyranjamie — APPROVED** [review]
Great work @fbwoolf, like the component generic 👍🏼

File: apps/mobile/src/common/transactions/bitcoin-transactions.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export interface GenerateBtcUnsignedTransactionCallbackArgs {
+  feeRate: number;
+  isSendingMax?: boolean;
+  utxos: CoinSelectionUtxo[];
+  values: BitcoinTransactionValues;
+}
+
+export function useGenerateBtcUnsignedTransactionNativeSegwit(
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/common/transactions/bitcoin-transactions.hooks.ts:37]: +1 this doesn't need to be a hook as it's not using any state. useCallback isn't needed if it's defined outside of react

[the remark this answers is not in the document]
Comment (fbwoolf) [reply apps/mobile/src/common/transactions/bitcoin-transactions.hooks.ts:37]: Good point, thanks, will change.

File: apps/mobile/src/features/send/send-form/components/send-form-amount-field.tsx [hunk 2]
```diff
@@ -3,15 +3,15 @@ import { Controller, useFormContext } from 'react-hook-form';
...
-export function SendFormAmountField() {
-  const { schema } = useSendFormContext();
+export function SendFormAmountField<T extends SendFormBaseContext<T>>() {
+  const { formData } = useSendFormContext<T>();
   const {
     control,
     // TODO: Handle errors
     // formState: { errors },
-  } = useFormContext<z.infer<typeof schema>>();
+  } = useFormContext<z.infer<typeof formData.schema>>();
```
Comment (pete-watters) [inline apps/mobile/src/features/send/send-form/components/send-form-amount-field.tsx:14]: Maybe we should add a task for handling the errors?

**Pull request merged**